src/pages/Graph/SummaryPanelEdge.tsx

```diff
@@ -252,12 +254,14 @@ export default class SummaryPanelEdge extends React.Component<SummaryPanelPropTy
         sourceLabel = 'source_workload';
         sourceValue = data.workload;
     }
-    const comparator = (metric: Metric) => {
-      if (this.isSpecialServiceDest(destMetricType)) {
-        return metric[sourceLabel] === sourceValue && metric.destination_workload === UNKNOWN;
-      }
-      return metric[sourceLabel] === sourceValue;
-    };
+    let comparator = (metric: Metric) => metric[sourceLabel] === sourceValue;
+    if (isServiceEntry) {
+      comparator = (metric: Metric) => {
+        return data.destServices && data.destServices.some(svc => svc.name === metric['destination_service_name']);
+      };
+    } else if (this.isSpecialServiceDest(destMetricType)) {
+      comparator = (metric: Metric) => metric[sourceLabel] === sourceValue && metric.destination_workload === UNKNOWN;
+    }
```

Comment:
So, if `isServiceEntry` is truthy, the evaluation of the `switch` is vain.

For code clarity, I'd prefer something like

```tsx
if (isServiceEntry) {
  /* all the old code here*/
} else {
  let comparator = (metric: Metric) => {
    return data.destServices && data.destServices.some(svc => svc.name === metric['destination_service_name']);
￼      return metric[sourceLabel] === sourceValue;
  };

  return getDatapoints(m, comparator);
}
```

It's just a small suggestion. It would be clearer for me, but for others it may be cristal clear as is. So, you choose to keep as is or to change.

What I really would like is that you add a comment on top of `getNodeDataPoints` definition. I see that when `isServiceEntry` arg is truthy, this method ignores `sourceMetricType` and `destMetricType` args (even if non-optional) and also assumes that `data` arg is the destination node and `m` are outbound metrics of the source node. On the other hand, if `isServiceEntry` arg is falsy, all args are used, but it's assumed that `data` arg is the source node and `m` are inbound metrics of the destination node. By not respecting these assumptions, this method will return garbage.

It took me a while to understand these assumptions, and the fact that all usages are passing `sourceMetricType` and `destMetricType` args is confusing for me.

Reply:
I think I understand what you mean, I was just a bit confused because I think your code snippet should start with `if (!isServiceEntry) {` instead of `if (isServiceEntry) {` , right?

Reply:
Yes, that's what I meant.
Thanks for changing :)